app.py

```diff
@@ -1,50 +1,82 @@
 import os
 import time
 import gradio as gr
+import requests
 from dotenv import load_dotenv
-from griptape.chat_demo import Chat
+from griptape.chat_demo import Chat_Azure
+from griptape.chat_cloud import Chat_Cloud
+from griptape.chat_local import Chat_Local
 
+# Load the environment variables
 load_dotenv()
+
+# Get the port from the environment variables for Gradio (default is 7860)
 port = os.getenv("GRADIO_PORT", 7860)
-chat = Chat()
 
+# Get the lambda endpoint from environment variables if using CDK based memory
+# If the lambda endpoint isn't specified, then the structure can ONLY be run LOCALLY.
+lambda_endpoint = os.getenv("LAMBDA_ENDPOINT", "")
+ 
+# Function to get the session id from the lambda endpoint (if not running locally).
+def get_session_id() -> str:
+    resp = requests.post(lambda_endpoint, json={"operation": "create_session"})
+    session_id = resp.json()["session_id"]
+    return session_id
+
+# Used for the Gradio ChatInterface. 
+# This adds the user message to the history and returns a tuple with an empty string and the history. 
 def user(user_message, history):
-    return "", history + [[user_message, None]]
+    history.append([user_message, None])
```

Comment:
I ended up removing this function because I found that only the bot function was necessary. But it's expecting a list. Once again, this was code already in the app.py that I didn't write so I'm not sure exactly what it's purpose was.